toolchain/lower/file_context.cpp

```diff
@@ -260,25 +258,29 @@ auto FileContext::CoalesceEquivalentSpecifics() -> void {
         }
       }
     }
+
+    // Update equivalent specific llvm::Functions, once all equivalences are
+    // found and the algorithm has converged on a canonical specific.
+    for (auto specific_id : specifics_to_delete) {
+      UpdateEquivalentSpecific(specific_id);
+      UpdateAndDeleteLLVMFunction(specific_id);
+    }
   }
 }
 
 auto FileContext::ProcessSpecificEquivalence(
-    std::pair<SemIR::SpecificId, SemIR::SpecificId> pair,
-    llvm::SmallVector<SemIR::SpecificId>& specifics_to_delete) -> void {
+    std::pair<SemIR::SpecificId, SemIR::SpecificId> pair) -> void {
   auto [specific_id1, specific_id2] = pair;
   CARBON_CHECK(specific_id1.has_value() && specific_id2.has_value(),
                "Expected values in equivalence check");
 
   auto get_canon = [&](SemIR::SpecificId specific_id) {
     return equivalent_specifics_.Get(specific_id).has_value()
-               ? std::make_pair(
-                     equivalent_specifics_.Get(specific_id),
-                     (equivalent_specifics_.Get(specific_id) != specific_id))
-               : std::make_pair(specific_id, false);
+               ? equivalent_specifics_.Get(specific_id)
+               : specific_id;
```

Comment:
The repeated `.Get` might be optimized away, but still feels nice to avoid. :)

```suggestion
    auto equiv_id = equivalent_specifics_.Get(specific_id);
    return equiv_id.has_value() ? equiv_id : specific_id;
```